### PR TITLE
feat: Implemented run app on runtime, start nodes on runtime, stop nodes on app quit, info fixes

### PR DESCRIPTION
**Pull request by alenmestrov**
**Issue Reference**  
Closes [#566](https://github.com/calimero-network/core/issues/566)

**Summary**  
This PR introduces several new features and improvements, including a new application menu, options for running the app and nodes on OS startup, proper cleanup of running nodes on app quit, and frontend enhancements for node configuration updates. These additions improve the application's usability and stability by providing more control over how it runs and manages node processes.

**Changes**  
- **New Menu Options**: Added a menu with the following options:
  - **Run on OS Startup**: Users can select this option to enable the application to start automatically when the operating system boots.
  - **Quit App**: Added an option to quit the application from the menu.
- **Run App on OS Startup**: Implemented functionality that allows the application to run automatically on OS startup when the "Run on OS Startup" option is enabled.
- **Run Nodes on App Startup**: Added a feature that allows users to specify which nodes should automatically start when the application launches.
- **Cleanup on App Quit**: Ensured that when the user quits the application, all running nodes are automatically stopped to prevent processes from lingering.
- **Node Running Status Check**: Fixed the logic for checking if a node process is running, ensuring that the application correctly detects node states.
- **Success Message on Node Configuration Update**: Improved the frontend to display a success message when a node's configuration is successfully updated.

**Testing**  
- Verified that the menu options work correctly:
  - The app starts on OS boot if the "Run on OS Startup" option is selected.
  - The app quits when the "Quit App" option is clicked.
- Tested that nodes marked to run on app startup successfully start when the application launches.
- Confirmed that all running nodes are properly stopped when the app is quit, and no processes remain active.
- Ensured the node running status check correctly identifies whether a node is active or stopped.
- Verified that the success message is displayed on the frontend when a node's configuration is updated.

**How to Test**  
1. Run the following commands:
   ```bash
   pnpm install
   pnpm run tauri dev
   ```
2. Open the application and check the new menu for:
   - "Run on OS Startup" and ensure selecting it enables the app to run on system startup.
   - "Quit App" and ensure it closes the application.
3. Set up nodes to run on app startup and restart the app to verify they automatically start when the app launches.
4. Quit the application and confirm that all running nodes are properly stopped.
5. Check the node status functionality to verify the correct state (running/stopped) is displayed.
6. Update a node's configuration and confirm that a success message is shown on the frontend.

### CODE REVIEW

File: src/pages/Dashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import { listen } from '@tauri-apps/api/event';
+
+interface TriggerActionPayload {
+  nodeName: string;
+  section: 'config' | 'logs' | 'controls';
+  action: string;
+}
+
+export interface TrayAction {
+  section: 'config' | 'logs' | 'controls';
```
Comment (frdomovic) [inline src/pages/Dashboard.tsx:19]: this is repeated, extract to enum or type

File: src/pages/Dashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    };
+
+    setupListeners();
+
+    return () => {
+      listeners.forEach(unlisten => unlisten());
+    };
+  }, []);
+
+  console.log('Dashboard rendered', selectedNode);
```
Comment (frdomovic) [inline src/pages/Dashboard.tsx:66]: remove

File: src/pages/Dashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        <NodeInitializationPopup
+          onInitialize={handleNodeInitialize}
+          onClose={() => setShowPopup(false)}
+        />
+      </PopupWrapper>
+    </DashboardContainer>
+  );
+};
+
+export default Dashboard;
```
Comment (frdomovic) [inline src/pages/Dashboard.tsx:111]: EOF

File: src/styles/NodeLogsStyles.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  color: #ffffff;
+  padding: 10px;
+  border-radius: 5px;
+  font-size: 12px;
+  font-family: monospace;
+  height: 300px;
+  overflow-y: auto;
+  white-space: pre-wrap;
+  word-wrap: break-word;
+`;
```
Comment (frdomovic) [inline src/styles/NodeLogsStyles.ts:30]: EOF

File: src/styles/SelectedNodeOperationsStyles.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    justify-content: center;
+  }
+
+  @media (max-width: 480px) {
+    flex-direction: column;
+    align-items: stretch;
+  }
+`;
+
+export const MainContent = styled.div`
```
Comment (frdomovic) [inline src/styles/SelectedNodeOperationsStyles.ts:30]: Regarding styles, if you update in first PR and rebase should be okay

**Review by frdomovic — APPROVED** [review]
LGTM, check comments

**Review by miraclx — APPROVED** [review]

**Pull request merged**